### PR TITLE
omni_types migrate into bebone

### PR DESCRIPTION
**Pull request by Maksasj**

### CODE REVIEW

File: examples/gfx/begui/0_begui_node_graph/imgui_node_graph_test.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #include <math.h>
 
-#define OMNI_TYPES_MATRIX_COLLUM_MAJOR_ORDER
-#define OMNI_TYPES_MATRIX4X4_PROJECTION_MATRIX_INVERSE_Y_AXIS
+#define BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline examples/gfx/begui/0_begui_node_graph/imgui_node_graph_test.h:3]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: examples/gfx/opengl/3_opengl_3d_cube/main.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #include <vector>
 
-#define OMNI_TYPES_MATRIX_COLLUM_MAJOR_ORDER
-#define OMNI_TYPES_MATRIX4X4_PROJECTION_MATRIX_INVERSE_Y_AXIS
+#define BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline examples/gfx/opengl/3_opengl_3d_cube/main.cpp:3]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: examples/gfx/opengl/4_opengl_bexel/bebone.h [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #ifndef BEBONE_H
 #define BEBONE_H
 
-#define OMNI_TYPES_MATRIX_COLLUM_MAJOR_ORDER
-// #define OMNI_TYPES_MATRIX4X4_PROJECTION_MATRIX_INVERSE_Y_AXIS
+#define BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline examples/gfx/opengl/4_opengl_bexel/bebone.h:4]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: examples/gfx/opengl/5_opengl_flappy_bird/main.cpp [hunk 4]
```diff
@@ -1,5 +1,5 @@
-#define OMNI_TYPES_MATRIX_COLLUM_MAJOR_ORDER
-#define OMNI_TYPES_MATRIX4X4_PROJECTION_MATRIX_INVERSE_Y_AXIS
+#define BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline examples/gfx/opengl/5_opengl_flappy_bird/main.cpp:1]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: examples/gfx/opengl/6_opengl_imgui/main.cpp [hunk 5]
```diff
@@ -1,7 +1,7 @@
 #include <vector>
 
-#define OMNI_TYPES_MATRIX_COLLUM_MAJOR_ORDER
-#define OMNI_TYPES_MATRIX4X4_PROJECTION_MATRIX_INVERSE_Y_AXIS
+#define BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline examples/gfx/opengl/6_opengl_imgui/main.cpp:3]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: examples/gfx/vulkan/2_vulkan_3d_cube/main.cpp [hunk 6]
```diff
@@ -1,5 +1,5 @@
-#define OMNI_TYPES_MATRIX_COLLUM_MAJOR_ORDER
-#define OMNI_TYPES_MATRIX4X4_PROJECTION_MATRIX_INVERSE_Y_AXIS
+#define BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline examples/gfx/vulkan/2_vulkan_3d_cube/main.cpp:1]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: src/bebone/core/types/color/color.tpp [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+#include "color_format.h"
+#include <glad/glad.h>
+
+namespace bebone::core {
+    template<ColorFormat _Format>
+    struct Color {
+
+    };
+}
```
Comment (Soskar1) [inline src/bebone/core/types/color/color.tpp:20]: I don't understand the purpose of this struct. Can you explain it for me?

File: src/bebone/core/types/utils/trait_bryan_angles.h [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+namespace bebone::core {
+    static inline Matrix<f32, 4, 4> trait_bryan_angle_xzy(const Vec3f& rotation) {
+        const f32 c3 = std::cos(rotation.z);
+        const f32 s3 = std::sin(rotation.z);
+        const f32 c2 = std::cos(rotation.x);
+        const f32 s2 = std::sin(rotation.x);
+        const f32 c1 = std::cos(rotation.y);
+        const f32 s1 = std::sin(rotation.y);
+
+        #ifndef BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline src/bebone/core/types/utils/trait_bryan_angles.h:15]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: src/bebone/core/types/utils/trait_bryan_angles.h [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+
+    static inline Matrix<f32, 4, 4> trait_bryan_angle_xyz(const Vec3f& rotation) {
+        const f32 c3 = std::cos(rotation.z);
+        const f32 s3 = std::sin(rotation.z);
+        const f32 c2 = std::cos(rotation.x);
+        const f32 s2 = std::sin(rotation.x);
+        const f32 c1 = std::cos(rotation.y);
+        const f32 s1 = std::sin(rotation.y);
+
+        #ifndef BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline src/bebone/core/types/utils/trait_bryan_angles.h:40]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: src/bebone/core/types/utils/trait_bryan_angles.h [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+
+    static inline Matrix<f32, 4, 4> trait_bryan_angle_yxz(const Vec3f& rotation) {
+        const f32 c3 = std::cos(rotation.z);
+        const f32 s3 = std::sin(rotation.z);
+        const f32 c2 = std::cos(rotation.x);
+        const f32 s2 = std::sin(rotation.x);
+        const f32 c1 = std::cos(rotation.y);
+        const f32 s1 = std::sin(rotation.y);
+
+        #ifndef BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline src/bebone/core/types/utils/trait_bryan_angles.h:65]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: src/bebone/core/types/utils/trait_bryan_angles.h [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+
+    static inline Matrix<f32, 4, 4> trait_bryan_angle_yzx(const Vec3f& rotation) {
+        const f32 c3 = std::cos(rotation.z);
+        const f32 s3 = std::sin(rotation.z);
+        const f32 c2 = std::cos(rotation.x);
+        const f32 s2 = std::sin(rotation.x);
+        const f32 c1 = std::cos(rotation.y);
+        const f32 s1 = std::sin(rotation.y);
+
+        #ifndef BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline src/bebone/core/types/utils/trait_bryan_angles.h:90]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: src/bebone/core/types/utils/trait_bryan_angles.h [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+    static inline Matrix<f32, 4, 4> trait_bryan_angle_zyx(const Vec3f& rotation) {
+        const f32 c3 = std::cos(rotation.z);
+        const f32 s3 = std::sin(rotation.z);
+        const f32 c2 = std::cos(rotation.x);
+        const f32 s2 = std::sin(rotation.x);
+        const f32 c1 = std::cos(rotation.y);
+        const f32 s1 = std::sin(rotation.y);
+
+        #ifndef BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline src/bebone/core/types/utils/trait_bryan_angles.h:115]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

File: src/bebone/core/types/utils/trait_bryan_angles.h [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+    static inline Matrix<f32, 4, 4> trait_bryan_angle_zxy(const Vec3f& rotation) {
+        const f32 c3 = std::cos(rotation.z);
+        const f32 s3 = std::sin(rotation.z);
+        const f32 c2 = std::cos(rotation.x);
+        const f32 s2 = std::sin(rotation.x);
+        const f32 c1 = std::cos(rotation.y);
+        const f32 s1 = std::sin(rotation.y);
+
+        #ifndef BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER
```
Comment (Soskar1) [inline src/bebone/core/types/utils/trait_bryan_angles.h:140]: Grammar error
Change `BEBONE_TYPES_MATRIX_COLLUM_MAJOR_ORDER` to `BEBONE_TYPES_MATRIX_COLUMN_MAJOR_ORDER`

**Review by Soskar1 — COMMENTED** [review]
Fix grammar errors please

File: examples/gfx/begui/0_begui_node_graph/imgui_node_graph_test.h (hunk 1 above)
Comment (Maksasj) [reply examples/gfx/begui/0_begui_node_graph/imgui_node_graph_test.h:3]: yes sir !

File: examples/gfx/opengl/3_opengl_3d_cube/main.cpp (hunk 2 above)
Comment (Maksasj) [reply examples/gfx/opengl/3_opengl_3d_cube/main.cpp:3]: yes sir !

File: examples/gfx/opengl/4_opengl_bexel/bebone.h (hunk 3 above)
Comment (Maksasj) [reply examples/gfx/opengl/4_opengl_bexel/bebone.h:4]: yes sir !

File: examples/gfx/opengl/5_opengl_flappy_bird/main.cpp (hunk 4 above)
Comment (Maksasj) [reply examples/gfx/opengl/5_opengl_flappy_bird/main.cpp:1]: yes sir !

File: examples/gfx/opengl/6_opengl_imgui/main.cpp (hunk 5 above)
Comment (Maksasj) [reply examples/gfx/opengl/6_opengl_imgui/main.cpp:3]: yes sir !

File: src/bebone/core/types/utils/trait_bryan_angles.h (hunk 9 above)
Comment (Maksasj) [reply src/bebone/core/types/utils/trait_bryan_angles.h:40]: yes sir !

File: src/bebone/core/types/utils/trait_bryan_angles.h (hunk 10 above)
Comment (Maksasj) [reply src/bebone/core/types/utils/trait_bryan_angles.h:65]: yes sir !

File: src/bebone/core/types/utils/trait_bryan_angles.h (hunk 11 above)
Comment (Maksasj) [reply src/bebone/core/types/utils/trait_bryan_angles.h:90]: yes sir !

**Review by Soskar1 — APPROVED** [review]
Great job! You can merge it safely

**Pull request merged**